src/routes/safes/safes.controller.spec.ts

```diff
@@ -0,0 +1,646 @@
+import { INestApplication } from '@nestjs/common';
+import {
+  fakeConfigurationService,
+  TestConfigurationModule,
+} from '../../config/__tests__/test.configuration.module';
+import {
+  fakeCacheService,
+  TestCacheModule,
+} from '../../datasources/cache/__tests__/test.cache.module';
+import { Test, TestingModule } from '@nestjs/testing';
+import { DomainModule } from '../../domain.module';
+import {
+  mockNetworkService,
+  TestNetworkModule,
+} from '../../datasources/network/__tests__/test.network.module';
+import { DataSourceErrorFilter } from '../common/filters/data-source-error.filter';
+import { SafesModule } from './safes.module';
+import * as request from 'supertest';
+import { chainBuilder } from '../../domain/chains/entities/__tests__/chain.builder';
+import { safeBuilder } from '../../domain/safe/entities/__tests__/safe.builder';
+import { masterCopyBuilder } from '../../domain/chains/entities/__tests__/master-copy.builder';
+import { contractBuilder } from '../../domain/contracts/entities/__tests__/contract.builder';
+import { pageBuilder } from '../../domain/entities/__tests__/page.builder';
+import {
+  multisigTransactionBuilder,
+  toJson as multisigTransactionToJson,
+} from '../../domain/safe/entities/__tests__/multisig-transaction.builder';
+import {
+  ethereumTransactionBuilder,
+  toJson as ethereumTransactionToJson,
+} from '../../domain/safe/entities/__tests__/ethereum-transaction.builder';
+import { faker } from '@faker-js/faker';
+import {
+  erc721TransferBuilder,
+  toJson as erc721TransferToJson,
+} from '../../domain/safe/entities/__tests__/erc721-transfer.builder';
+import {
+  moduleTransactionBuilder,
+  toJson as moduleTransactionToJson,
+} from '../../domain/safe/entities/__tests__/module-transaction.builder';
+
+describe('Safes Controller (Unit)', () => {
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    fakeConfigurationService.set(
+      'safeConfig.baseUri',
+      'https://test.safe.config',
+    );
+
+    fakeConfigurationService.set(
+      'exchange.baseUri',
+      'https://test.exchange.service',
+    );
+
+    fakeConfigurationService.set('exchange.apiKey', 'https://test.api.key');
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        SafesModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  it('safe info is correctly serialised', async () => {
+    const chain = chainBuilder()
+      .with('recommendedMasterCopyVersion', '5.0.0')
```

Comment:
Here it's a little bit more tricky but I'll use faker to generate one version and reuse it across this test. The reason why I specified one was because if it's not the same one it generates a different response. Leaving it to a random generation would sometimes make the test pass or not (if the same semver is generated)